src/main/scala/io/iohk/ethereum/domain/Blockchain.scala

```diff
@@ -166,7 +166,7 @@ trait Blockchain {
 
   def saveBestKnownBlock(number: BigInt): Unit
 
-  def saveNode(nodeHash: NodeHash, nodeEncoded: NodeEncoded, blockNumber: BigInt): Unit
+  def saveNode(nodeHash: NodeHash, nodeEncoded: NodeEncoded, blockNumber: BigInt, withSnapshotSave: Boolean): Unit
```

Comment:
Although this solution will work, I'm not quite sure if this continues polluting upper layers with pruning internal detail. I'm wondering if we could configure FastSync with a storage that won't save snapshots and regular sync with one that will instead. Do you think it's too much? Maybe we work on a refactor on the next version

Reply:
I don't think it is too much, it would be beneficial to refactor it the way you say it. Especially that we already have tasks to refactor both fast and regular sync. I will leave task also for this refactoring.
During implementing this solution I was thinking about it but decided that last moment refactors are not best idea here.